vg/vg_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vg_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/plotter"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+	"github.com/gonum/plot/vg/vgeps"
+	"github.com/gonum/plot/vg/vgimg"
+	"github.com/gonum/plot/vg/vgpdf"
+	"github.com/gonum/plot/vg/vgsvg"
+)
+
+// TestLineWidth tests output against test images generated by
+// gen_width.go - these images have been visually inspected and
+// validated.
+func TestLineWidth(t *testing.T) {
+	formats := []string{
+		// TODO: Add logic to cope with run to run eps differences.
+		"pdf",
+		"svg",
+		"png",
+		"tiff",
+		"jpg",
+	}
+
+	const (
+		width  = 100
+		height = 100
+	)
+
+	for _, w := range []vg.Length{-1, 0, 1} {
+		for _, typ := range formats {
+			p, err := lines(w)
+			if err != nil {
+				log.Fatalf("failed to create plot for %v:%s: %v", w, typ, err)
+			}
+
+			var c interface {
+				vg.Canvas
+				Size() (w, h vg.Length)
+				io.WriterTo
+			}
+
+			name := fmt.Sprintf("width_%v.%s", w, typ)
+			switch typ {
```

Comment:
We need to open up an issue to get this logic into a method in plot so that we can replace plot.Save with plot.Write. Then this test can call that method instead.